s3_resource/src/s3_operations.cpp

```diff
@@ -2392,4 +2393,207 @@ namespace irods_s3 {
 		return SUCCESS();
 	} // s3_notify_operation
 
+    irods::error s3_read_checksum(irods::plugin_context& _ctx,
+            const std::string* _checksum_scheme,
+            std::string* _returned_checksum) {
+
+        // validate inputs are not null
+        if (!_checksum_scheme || !_returned_checksum) {
+            return ERROR(SYS_INVALID_INPUT_PARAM, "Invalid input parameter.");
```

Comment:
The error code reports that there was an invalid input. The error message can go a step further and state that a null pointer was detected. It doesn't need to state which pointer was null, but knowing a null pointer was detected would be good.

Reply:
done